src/ipa_pc/mod.rs

```diff
@@ -736,6 +736,148 @@ impl<G: AffineCurve, D: Digest> PolynomialCommitment<G::ScalarField> for InnerPr
+
+            let labeled_polynomial =
+                poly_map.get(label).ok_or(Error::MissingPolynomial {
+                    label: label.to_string(),
+                })?;
+
+            points.push(point);
+
+            // y_i
+            let evaluated_y = labeled_polynomial.polynomial().evaluate(*point); 
+
+            // (p_i(X) - y_i) / (X - x_i)
+            let polynomial = 
+                &(labeled_polynomial.polynomial() - &Polynomial::from_coefficients_vec(vec![evaluated_y])) 
+                / 
+                &Polynomial::from_coefficients_vec(vec![
+                    (G::ScalarField::zero() - point), 
+                    G::ScalarField::one()
+                ]);
+            
+            // h(X) = SUM( lambda^i * ((p_i(X) - y_i) / (X - x_i)) )
+            batch_polynomial += (cur_challenge, &polynomial);
+
+            // lambda^i
+            cur_challenge = opening_challenges(opening_challenge_counter);
+            opening_challenge_counter += 1;
+        }
+
+        // Commitment of the h(X) polynomial
+        let batch_commitment = Self::cm_commit(
+            ck.comm_key.as_slice(),
+            batch_polynomial.coeffs.as_slice(),
+            None,
+            None,
+        ).into_affine();
+
+        // TODO: When we will move to Sponge-based construction, we will absorb the batch_commitment only and then squeeze a new challenge
+        // Fresh random challenge x
+        let point= Self::compute_random_oracle_challenge(
+            &to_bytes![
+                batch_commitment,
+                points                
+            ]
+            .unwrap(),
+        );
+
+        // Values: v_i = p_i(x), where x is fresh random challenge
+        let batch_values: BTreeMap<_, _> = labeled_polynomials
+            .iter()
+            .map(|labeled_polynomial| (labeled_polynomial.label().clone(), labeled_polynomial.polynomial().evaluate(point)))
+            .collect();
+
+        // h(X) polynomial added to the set of polynomials for multi-poly single-point batching
+        let mut labeled_polynomials = labeled_polynomials;
+        let labeled_batch_polynomial = LabeledPolynomial::new(
+            format!("Batch"),
+            batch_polynomial,
+            None,
+            None
+        );
+        labeled_polynomials.push(&labeled_batch_polynomial);
+
+        // Commitment of h(X) polynomial added to the set of polynomials for multi-poly single-point batching
+        let mut commitments = commitments;
+        let labeled_batch_commitment = LabeledCommitment::new(
+            format!("Batch"),
+            Commitment { comm: batch_commitment, shifted_comm: None },
+            None
+        );
+        commitments.push(&labeled_batch_commitment);
+
+        let mut rands = rands;
+        let batch_randomness = Randomness::empty();
+        rands.push(&batch_randomness);
+
+        end_timer!(batch_time);
+
+        let opening_challenge = Self::compute_random_oracle_challenge(
```

Comment:
When not using a sponge, we need to hash all commitments along with the batch_commitment to the random oracle to produce the opening_challenge.

Reply:
Didn't we already address this with the TODO ?

Reply:
That is now issue of a future PR.